gnocchi/storage/incoming/redis.py

```diff
@@ -49,13 +48,27 @@ def _store_new_measures(self, metric, data):
         self._client.rpush(path, data)
 
     def _build_report(self, details):
+        build_metric_list = (lambda x: x.decode('utf8').split(redis.SEP)[1]
+                             if details else lambda x: True)
+        metrics = 0
+        measures = 0
         match = redis.SEP.join([self.get_sack_name("*"), "*"])
-        metric_details = collections.defaultdict(int)
-        for key in self._client.scan_iter(match=match, count=1000):
-            metric = key.decode('utf8').split(redis.SEP)[1]
-            metric_details[metric] = self._client.llen(key)
-        return (len(metric_details.keys()), sum(metric_details.values()),
-                metric_details if details else None)
+        metric_details = {}
+        cursor = '0'
```

Comment:
stolen from source[1] but basically redis returns '0' as cursor when it hits end of list[2]

[1] https://github.com/andymccurdy/redis-py/blob/d6c300b39ab4c81cd97b81fcd8d25a9cde9476e7/redis/client.py#L1469-L1482
[2] https://redis.io/commands/scan